Wire-iOS/Sources/UserInterface/Services/ServiceDetailViewController.swift

```
@@ -0,0 +1,210 @@
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+//
+// This program is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+// GNU General Public License for more details.
+//
+// You should have received a copy of the GNU General Public License
+// along with this program. If not, see http://www.gnu.org/licenses/.
+//
+
+
+import Foundation
+import Cartography
+
+public enum ServiceConversation {
+    case existing(ZMConversation)
+    case new
+}
+
+public struct Service {
+    let serviceUser: ServiceUser
+    var serviceUserDetails: ServiceDetails?
+    var provider: ServiceProvider?
+}
+
+extension Service {
+    init(serviceUser: ServiceUser) {
+        self.serviceUser = serviceUser
+        self.serviceUserDetails = nil
+        self.provider = nil
+    }
+}
+
+extension ServiceConversation: Hashable {
+    
+    public static func ==(lhs: ServiceConversation, rhs: ServiceConversation) -> Bool {
+        return lhs.hashValue == rhs.hashValue
+    }
+    
+    public var hashValue: Int {
+        switch self {
+        case .new:
+            return 0
+        case .existing(let conversation):
+            return conversation.hashValue
+        }
+    }
+}
+
+private func add(service: Service, to conversation: Any) {
+    guard let userSession = ZMUserSession.shared(),
+           let serviceConversation = conversation as? ServiceConversation else {
+        return
+    }
+    
+    switch serviceConversation {
+    case .new:
+        userSession.startConversation(with: service.serviceUser) { conversation in
+            
+        }
+    case .existing(let conversation):
+        conversation.add(serviceUser: service.serviceUser, in: userSession) { done in
```

Same here.

---

You are right, but this one is still WIP, we also need to report the correct error.